### ISSUE TITLE
Preferences can't be saved when running from jar

### ISSUE THREAD

**Issue by pont-us** [pont-us/PuffinPlot#326]
Version: ca242474a1 (current repository head).

Steps to reproduce: build and run the PuffinPlot jar. Open the preferences dialog, then close it with the "Close" button.

Expected behaviour: preferences dialog opens and closes without incident. Any changes made are saved for the next session.

Observed behaviour:  An error dialog with the message "The preferences could not be saved." is shown. Changes are not saved for the next session. The following error is printed to the standard error stream:

```
java.util.prefs.BackingStoreException: java.util.MissingResourceException: Could not load any resource bundles.org.apache.xalan.res.XSLTErrorResources
	at java.util.prefs.FileSystemPreferences$7.run(FileSystemPreferences.java:642)
```

[Full backtrace.](https://github.com/pont-us/PuffinPlot/files/5577331/backtrace.txt)

**Comment by pont-us** [comment]
This error was introduced with the switch from Ant to Maven in commit bb313d49 , and has gone unnoticed because no releases have been made since then. The cause is the `minimizeJar` option passed to the shade plugin which produces the fat jar (a.k.a. "uber jar") combining PuffinPlot and the necessary parts of third-party libraries. `minimizeJar` excludes unused classes, dramatically reducing the size of the jar, but has erroneously excluded `org.apache.xalan.res.XSLTErrorResources`, presumably because it's being dynamically loaded. It looks as though this particular error is generated during error reporting and is masking the actual cause, but it seems likely that the initial error might also be caused by a missing class in the jar.

**Comment by pont-us** [comment]
Closed by commit 8778659c6 , which updates the POM to add `org.apache.xalan.res.XSLTErrorResources` and
`org.apache.xml.serializer.ToXMLStream` explicitly to the jar (the second being the omission that triggers the initial error).
